packages/date-picker/test/helpers.js

```diff
@@ -108,24 +109,38 @@ export function getFirstVisibleItem(scroller, bufferOffset = 0) {
   });
 }
 
-export function getFocusedMonth(overlayContent) {
-  const months = [...overlayContent.querySelectorAll('vaadin-month-calendar')];
-  return months.find((month) => {
-    return !!month.shadowRoot.querySelector('[part~="focused"]');
+/**
+ * @param {HTMLElement} root vaadin-date-picker or vaadin-date-picker-overlay-content
+ */
+export function getFocusableCell(root) {
+  const overlayContent = root._overlayContent ?? root;
+  const focusableMonth = [...overlayContent.querySelectorAll('vaadin-month-calendar')].find((month) => {
+    return !!month.shadowRoot.querySelector('[tabindex="0"]');
   });
+
+  if (focusableMonth) {
+    return focusableMonth.shadowRoot.querySelector('[tabindex="0"]');
+  }
 }
 
-export function getFocusedCell(overlayContent) {
-  const focusedMonth = getFocusedMonth(overlayContent);
-  return focusedMonth.shadowRoot.querySelector('[part~="focused"]');
+/**
+ * @param {HTMLElement} root vaadin-date-picker or vaadin-date-picker-overlay-content
+ */
+export function getFocusedCell(root) {
+  const focusableCell = getFocusableCell(root);
+  if (focusableCell && isElementFocused(focusableCell)) {
+    return focusableCell;
+  }
 }
 
 /**
  * Waits for the scroll to finish in the date-picker overlay content.
  *
- * @param {HTMLElement} overlayContent
+ * @param {HTMLElement} root vaadin-date-picker or vaadin-date-picker-overlay-content
  */
-export async function waitForScrollToFinish(overlayContent) {
+export async function waitForScrollToFinish(root) {
```

Comment:
**note:** I also changed the signature of these methods to allow for passing `datePicker` instead of `datePicker._overlayContent` which slightly more convenient.